networkx/algorithms/community/label_propagation.py

```diff
@@ -64,31 +64,38 @@ def asyn_lpa_communities(G, weight=None, seed=None):
 
     labels = {n: i for i, n in enumerate(G)}
     cont = True
+
     while cont:
         cont = False
         nodes = list(G)
         seed.shuffle(nodes)
-        # Calculate the label for each node
+
         for node in nodes:
-            if len(G[node]) < 1:
+            adj = G[node]
+            if not adj:
                 continue
 
-            # Get label frequencies. Depending on the order they are processed
-            # in some nodes with be in t and others in t-1, making the
-            # algorithm asynchronous.
+            # Get label frequencies among adjacent nodes.
+            # Depending on the order they are processed in,
+            # some nodes will be in iteration t and others in t-1,
+            # making the algorithm asynchronous.
             label_freq = Counter()
-            for v in G[node]:
-                label_freq.update(
-                    {labels[v]: G.edges[node, v][weight] if weight else 1}
-                )
-            # Choose the label with the highest frecuency. If more than 1 label
-            # has the highest frecuency choose one randomly.
+            if weight is None:
+                label_freq.update(map(labels.get, adj))
+            else:
+                for v in adj:
+                    label_freq.update({labels[v]: G.edges[node, v][weight]})
```

Comment:
I see what you mean about `Counter` treating an iterator differently than I was expecting. It seems that Counter is slower than a dict for weighted counting.  We should be able to speed up what we've got (my tests with 2000 nodes give >3 times speedup after switching to a dict).
```python
label_freq={labels[v]:0 for v in G[node]}
for _, v, wt in G.edges(node, data=weight, default=1):
    label_freq[labels[v]] += wt
```

Reply:
Great point, the defaultdict approach is also much faster for me when using weighted counts. For unweighted counts it's still faster to initialise a Counter from an iterator/list, but as both are dict subclasses it shouldn't be an issue to just use a Counter if `weight=None` and a dict otherwise.